lib/oidc/key_service.rb

```diff
@@ -7,23 +7,43 @@ class KeyService
     @mutex = Mutex.new
     # Map from kid to OpenSSL::RSA::PKey for all current keys
     @current_keys = {}
-
-    class << self
-      attr_reader :current_keys
-    end
+    @cache_miss_kids = {}
+    KID_CACHE_PERIOD_SECS = 60
+    KID_CACHE_MAX_SIZE = 10
```

Comment:
what's the current rate of cache misses, approximately?

Reply:
Once this is deployed it should be about 1/day/instance, because there's only a single bad kid value that's being sent. Assuming all the clients are well behaved and that okta rotates the keys once/month there really should only ever be a maximum of one entry in here.

Reply:
Actually we've received two different invalid `kid` values in the past 24 hours. Some non-json lines in the logs were breaking my script and I hadn't noticed.

Reply:
`awslogs get -GS -s 24h -f 'Public key' dsva-vagov-prod/srv/vets-api/src/log/vets-api-server.log | jq -R '. as $raw | try (fromjson | .payload.kid) catch $raw' | sort | uniq -c`